### ISSUE TITLE
Dream of the Red Chamber files crash Tokenizer

### ISSUE THREAD

**Issue by scottkleinman** [WheatonCS/Lexos#441]
Transferred from issue #329.

The _Dream of the Red Chamber_ files in the Rolling Windows TestSuite experiment cause Tokenizer to crash. The table is not drawn, but no error is given in either the Python or browser console. The problem does not occur with any other Chinese text tested--only the files in the `红楼梦HongLouMeng_Dream_of_the_Red Chamber` folder. Is there something unusual about the encoding of these files?

**Comment by scottkleinman** [comment]
I ran `红楼梦HongLouMeng_Dream_of_the_Red Chamber/FilesToUse/Break by Chapter 80/Chapter64` through Scrubber, downloaded it, and then uploaded the scrubbed version. With this version it actually worked.

I suggest that we scrub the entire folder with something like "Make Lowercase" (which should have no effect on a Chinese text) and replace the current folder with the new one.

**Comment by chantisnake** [comment]
I will try to investigate on this when I have time...

**Comment by chantisnake** [comment]
I think we need to know what causes the bug before deciding how to fix it.

**Comment by scottkleinman** [comment]
I thought it might be a BOM issue, but the current files are encoded without the BOM.

**Comment by chantisnake** [comment]
chardet will be able to detect BOM I think... That is basic.

I will work on this now.

<s>what a great world it will be if the encoding always has BOM...</s>

**Comment by chantisnake** [comment]
I just did a debug on tokenize. The code of tokenize `GET` is extremely ugly and confusing, but I think everything sent to the front end is correct. 

but for some reason the ajax call is never triggered to post... (I think there should be a `POST` automatically triggered by get after we send a `GET`)

I think this is mainly a front end issue.

**Comment by scottkleinman** [comment]
I've discovered the problem. The _Dream of the Red Chamber_ files have straight (i.e. non-curly) quotation marks that are not being escaped. So this will presumably affect other files with straight quotation marks. We'll need to figure out how to escape them.

**Comment by chantisnake** [comment]
Yes, I was just  about to mention that in `GenAB` my table is also not rendered, I think these two maybe are related, you can add quotes to excape HTML function and then apply that function to all the words.

**Comment by scottkleinman** [comment]
Hmmm... Here's the `html_escape` function table:

```python
html_escape_table = {
    u"&": u"&amp;",
    u'"': u"&quot;",
    u"'": u"&apos;",
    u">": u"&gt;",
    u"<": u"&lt;",
}
```
So unless that comma at the end is breaking the function, it should already work. It may need to be applied deeper in the back end when the matrix is generated.

**Comment by chantisnake** [comment]
I think adding that is exactly how to use it.

Do you want me to fix it, or you are okay all by yourself?

**Comment by scottkleinman** [comment]
That's already in the function. So it's really a question of where to apply the function. This is lower priority than some of the other tasks I'm on, so if you feel you have time, go ahead and fix it. Otherwise, I'll get to it eventually. But, I think your first priority might be to get the session variable for local mode work, as that will make your life much easier...

**Comment by chantisnake** [comment]
fixed in commit  25d3f84
